pkgs/os-specific/linux/trace-cmd/default.nix

```diff
@@ -1,25 +1,39 @@
 { stdenv, fetchgit, asciidoc, docbook_xsl, libxslt }:
-
-stdenv.mkDerivation rec {
+let
+  srcSpec = import ./src.nix;
+  shortRev = builtins.substring 0 7 srcSpec.rev;
+in stdenv.mkDerivation rec {
   name    = "trace-cmd-${version}";
-  version = "2.6";
+  version = "2.8-${shortRev}";
 
-  src = fetchgit {
-    url    = "git://git.kernel.org/pub/scm/linux/kernel/git/rostedt/trace-cmd.git";
-    rev    = "refs/tags/trace-cmd-v${version}";
-    sha256 = "15d6b7l766h2mamqgphx6l6a33b1zn0yar2h7i6b24ph6kz3idxn";
-  };
+  src = fetchgit srcSpec;
+
+  patches = [ ./fix-Makefiles.patch ];
 
   buildInputs = [ asciidoc libxslt ];
 
-  configurePhase = "true";
-  buildPhase     = "make prefix=$out MANPAGE_DOCBOOK_XSL=${docbook_xsl}/xml/xsl/docbook/manpages/docbook.xsl all doc";
-  installPhase   = "make prefix=$out install install_doc";
+  outputs = [ "out" "lib" "dev" "man" ];
+
+  MANPAGE_DOCBOOK_XSL="${docbook_xsl}/xml/xsl/docbook/manpages/docbook.xsl";
+
+  buildPhase = "make trace-cmd libs doc";
```

Comment:
Please use

```nix
buildFlags = [ "trace-cmd" "libs" "doc" ];
```

Overriding phases like this breaks hooks and other features of the generic builder.

Reply:
I tried to build with either `buildFlags = [ "trace-cmd" "libs" "doc" ];` or `makeFlags = [ "trace-cmd" "libs" "doc" ];` but both result in a failing build.

Reply:
It seems to be caused by generic builder adding `SHELL` variable:

https://github.com/NixOS/nixpkgs/blob/9dd1bbc9f39fced89aef15de3cc4f50da1fca3d6/pkgs/stdenv/generic/setup.sh#L1027

Which is extremely weird, since the makefiles do not seem to refer to `SHELL`. And even weider, the linking error is from not finding a symbol that is in the same file:

```
/nix/store/1kl6ms8x56iyhylb2r83lq7j3jbnix7w-binutils-2.31.1/bin/ld: /build/trace-cmd-a8faf36/tracecmd/trace-record.o: in function `reset_event_pid':
/build/trace-cmd-a8faf36/tracecmd/trace-record.c:3850: undefined reference to `add_event_pid'
collect2: error: ld returned 1 exit status
```